tests/test_bootstrap.py

```diff
@@ -0,0 +1,1390 @@
+            plt.xlim([0, 1])
+            plt.show()
+
+        np.testing.assert_allclose(J_dot_B_Redl[1:-1], J_dot_B_sfincs[1:-1], rtol=0.1)
+
+
+class TestBootstrapObjectives:
+    """Tests for bootstrap current objective functions."""
+
+    @pytest.mark.unit
+    def test_BootstrapRedlConsistency_12(self):
+        """Test for bootstrap current in vacuum field.
+
+        The BootstrapRedlConsistency objective function should be
+        1/2 if the equilibrium is a vacuum field.
+        """
+        helicity = (1, -4)
+        filename = ".//tests//inputs//LandremanPaul2022_QH_reactorScale_lowRes.h5"
+        eq = desc.io.load(filename)
+        # n and T profiles do not go to 0 at rho=0 to avoid divide-by-0 there:
+        eq.electron_density = PowerSeriesProfile(
+            5.0e20 * np.array([1, -0.9]), modes=[0, 8]
+        )
+        eq.electron_temperature = PowerSeriesProfile(
+            8e3 * np.array([1, -0.9]), modes=[0, 2]
+        )
+        eq.ion_temperature = PowerSeriesProfile(9e3 * np.array([1, -0.9]), modes=[0, 2])
+        eq.atomic_number = PowerSeriesProfile(np.array([1.2, 0.2]), modes=[0, 2])
+
+        def test(grid_type, kwargs, L_factor, M_factor, N_factor, rho_exponent):
+            grid = grid_type(
+                L=eq.L * L_factor,
+                M=eq.M * M_factor,
+                N=eq.N * N_factor,
+                NFP=eq.NFP,
+                **kwargs,
+            )
+            obj = ObjectiveFunction(
+                BootstrapRedlConsistency(
+                    grid=grid,
+                    helicity=helicity,
+                    rho_exponent=rho_exponent,
+                ),
+                eq,
+            )
+            scalar_objective = obj.compute_scalar(obj.x(eq))
+            print("Should be approximately 0.5:", scalar_objective)
+            return scalar_objective
+
+        results = []
+
+        # Loop over grid trypes. For LinearGrid we need to drop the
+        # point at rho=0 to avoid a divide-by-0
+        grid_types = [LinearGrid, QuadratureGrid]
+        kwargs = [{"axis": False}, {}]
+
+        # Loop over grid resolutions:
+        L_factors = [1, 2, 1, 1, 1]
+        M_factors = [1, 1, 2, 1, 1]
+        N_factors = [1, 1, 1, 2, 1]
+        rho_exponents = [0, 1, 1, 1, 2]
+
+        for grid_type, kwargs in zip(grid_types, kwargs):
+            for L_factor, M_factor, N_factor, rho_exponent in zip(
+                L_factors, M_factors, N_factors, rho_exponents
+            ):
+                results.append(
+                    test(grid_type, kwargs, L_factor, M_factor, N_factor, rho_exponent)
+                )
+
+        np.testing.assert_allclose(np.array(results), 0.5, rtol=2e-4)
+
+    @pytest.mark.unit
+    def test_BootstrapRedlConsistency_resolution(self):
+        """Confirm that the objective function is ~independent of grid resolution."""
+        helicity = (1, 0)
+        filename = ".//tests//inputs//DSHAPE_output_saved_without_current.h5"
```

Comment:
might be better to use the `DSHAPE` test fixture here, which is re-run each time to make sure changes to the rest of the code don't break the bootstrap stuff.

Reply:
I attempted to address this, copying the pattern from the other test files.